test/libsolidity/SemanticTest.cpp

```diff
@@ -0,0 +1,163 @@
+	{
+		bytes output = callContractFunctionWithValueNoEncoding(
+			test.call.signature,
+			test.call.value,
+			test.call.arguments.rawBytes
+		);
+
+		if ((m_transactionSuccessful != test.call.expectations.status) || (output != test.call.expectations.rawBytes))
+			success = false;
+
+		string resultOutput;
+		if (m_transactionSuccessful)
+			resultOutput = "-> " + TestFileParser::bytesToFormattedString(output, test.call.expectations.formats);
+		else
+			resultOutput = "REVERT";
+
+		test.status = m_transactionSuccessful;
+		test.rawBytes = std::move(output);
+		test.output = std::move(resultOutput);
+	}
+
+	if (!success)
+	{
+		string nextIndentLevel = _linePrefix + "  ";
+		FormattedScope(_stream, _formatted, {BOLD, CYAN}) << _linePrefix << "Expected result:" << endl;
+		for (auto const& test: m_tests)
+		{
+			printFunctionCall(_stream, test.call, _linePrefix);
+			printFunctionCallTest(_stream, test, true, _linePrefix, _formatted);
+		}
+
+		FormattedScope(_stream, _formatted, {BOLD, CYAN}) << _linePrefix << "Obtained result:" << endl;
+		for (auto const& test: m_tests)
+		{
+			printFunctionCall(_stream, test.call, _linePrefix);
+			printFunctionCallTest(_stream, test, false, _linePrefix, _formatted);
+		}
+		return false;
+	}
+	return true;
+}
+
+void SemanticTest::printSource(ostream& _stream, string const& _linePrefix, bool const) const
+{
+	stringstream stream(m_source);
+	string line;
+	while (getline(stream, line))
+		_stream << _linePrefix << line << endl;
+}
+
+void SemanticTest::printUpdatedExpectations(ostream& _stream, string const& _linePrefix) const
+{
+	for (auto const& test: m_tests)
+	{
+		printFunctionCall(_stream, test.call, _linePrefix);
+		printFunctionCallTest(_stream, test, false, _linePrefix);
+	}
+}
+
+
+void SemanticTest::parseExpectations(istream& _stream)
+{
+	TestFileParser parser{_stream};
+	for (auto const& call: parser.parseFunctionCalls())
+		m_tests.emplace_back(FunctionCallTest{std::move(call), false, bytes{}, string{}});
+}
+
+bool SemanticTest::deploy(string const& _contractName, u256 const& _value, bytes const& _arguments)
+{
+	auto output = compileAndRunWithoutCheck(m_source, _value, _contractName, _arguments, m_libraryAddresses);
+	return !output.empty() && m_transactionSuccessful;
+}
+
+void SemanticTest::printFunctionCall(ostream& _stream, FunctionCall const& _call, string const& _linePrefix) const
+{
+	_stream << _linePrefix << _call.signature;
+	if (_call.value > u256(0))
+		_stream << "[" << _call.value << "]";
+	if (!_call.arguments.raw.empty())
+		_stream << ": " << boost::algorithm::trim_copy(_call.arguments.raw);
+	if (!_call.arguments.comment.empty())
+		_stream << " # " << _call.arguments.comment;
+	_stream << endl;
+}
+
+void SemanticTest::printFunctionCallTest(
+	ostream& _stream,
+	FunctionCallTest const& _test,
+	bool _expected,
+	string const& _linePrefix,
+	bool const _formatted
+) const
+{
+	_stream << _linePrefix;
+	if (_formatted && !_test.matchesExpectation())
```

Comment:
I wouldn't have expected this functionality from the name. What about `printFunctionCallFormatted` or `Highlighted`?

Reply:
Updated the function name and its documentation.